File: user-service/src/main/java/rs/raf/user_service/domain/entity/ActuaryLimit.java
package rs.raf.user_service.domain.entity;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.RequiredArgsConstructor;

import javax.persistence.*;
import java.math.BigDecimal;

@Data
@Entity(name = "actuary_limits")
@AllArgsConstructor
@RequiredArgsConstructor
public class ActuaryLimit {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private BigDecimal limitAmount;
    private BigDecimal usedLimit;
    private boolean needsApproval;
    @OneToOne
    private Employee employee;

    public ActuaryLimit(BigDecimal limitAmount, BigDecimal usedLimit, boolean needsApproval, Employee employee) {
        this.limitAmount = limitAmount;
        this.usedLimit = usedLimit;
        this.needsApproval = needsApproval;
        this.employee = employee;
    }
}

File: user-service-kotlin/src/main/kotlin/pack/userservicekotlin/domain/entities/ActuaryLimit.kt
package pack.userservicekotlin.domain.entities

import jakarta.persistence.*
import java.math.BigDecimal

@Entity(name = "actuary_limits")
open class ActuaryLimit(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    open var id: Long? = null,
    open var limitAmount: BigDecimal? = null,
    open var usedLimit: BigDecimal? = null,
    open var needsApproval: Boolean = false,
    @OneToOne
    open var employee: Employee? = null,
)
